src/test/java/BaseTest.java

```diff
@@ -1,18 +1,169 @@
+            }
+        }
+
+    public WebDriver lambdaTest() throws MalformedURLException {
+
+        String hubURL = "https://hub.lambdatest.com/wd/hub";
+
+        DesiredCapabilities capabilities = new DesiredCapabilities();
+        capabilities.setCapability("browserName", "Chrome");
+        capabilities.setCapability("browserVersion", "108.0");
+        HashMap<String, Object> ltOptions = new HashMap<String, Object>();
+        ltOptions.put("user", "shalinibaronia");
+        ltOptions.put("accessKey", "fPH3tVYv2I2ZMOQGbtMgkKRXMVoMIN4ya76moFRzQkigGFIOn8");
+        ltOptions.put("build", "Selenium 4");
+        ltOptions.put("name", this.getClass().getName());
+        ltOptions.put("platformName", "Windows 11");
+        ltOptions.put("seCdp", true);
+        ltOptions.put("selenium_version", "4.0.0");
+        capabilities.setCapability("LT:Options", ltOptions);
+        return new RemoteWebDriver(new URL(hubURL), capabilities);
+    }
 
-    @BeforeSuite
-    static void setupClass() {
-        WebDriverManager.chromedriver().setup();
+    public void login(String email, String password) {
+        provideEmail(email);
+        providePassword(password);
+        clickSubmit();
     }
 
-}
+    public void clickSubmit() {
+        WebElement submitButton = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("button[type='submit']")));
+        submitButton.click();
+    }
+
+    public void providePassword(String password) {
+        WebElement passwordField = getDriver().findElement(By.cssSelector("[type='password']"));
+        wait.until(ExpectedConditions.elementToBeClickable(passwordField));// use this when method only take WebElement
+
+        passwordField.clear();
+        passwordField.sendKeys(password);
+    }
+
+    public void provideEmail(String email) {
+        WebElement emailField = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[type='email']")));
+        emailField.clear();
+        emailField.sendKeys(email);
+    }
+
+
+    public void clickAvatarIcon() {
+        WebElement avatarIcon = getDriver().findElement(By.cssSelector("img.avatar"));
+        avatarIcon.click();
+
+    }
+
+    public void provideCurrentPassword(String password) {
+        WebElement currentPassword = getDriver().findElement(By.cssSelector("[name='current_password']"));
+        currentPassword.clear();
+        currentPassword.sendKeys(password);
+    }
+
+    public String generateRandomName() {
+        return UUID.randomUUID().toString().replace("-", "");//
+    }
+
+    public void clickSaveButton() {
+        WebElement saveButton = getDriver().findElement(By.cssSelector("button.btn-submit"));
+        saveButton.click();
+    }
+
+    public void provideProfileName(String randomName) {
+        WebElement profileName = getDriver().findElement(By.cssSelector("[name='name']"));
+        profileName.clear();
+        profileName.sendKeys(randomName);
+    }
```

Comment:
Base Test in the main class which will take care of Browser setup, loading configuration file and other reusable methods like screenshot, handling sync issues and many more.

Please move this methods in other classes (Using POM structure allows you to write tests using page actions without having to worry about the logic behind them)

For example

![image](https://user-images.githubusercontent.com/104761482/216757326-e5da4731-0f0f-492f-9eea-785fe7568c02.png)